python/lsst/resources/_resourceHandles/_httpResourceHandle.py

```diff
@@ -129,27 +140,64 @@ def read(self, size: int = -1) -> bytes:
 
             return self._completeBuffer.getbuffer().tobytes()
 
-        # a partial read is required, either because a size has been specified,
-        # or a read has previously been done.
+        # A partial read is required, either because a size has been specified,
+        # or a read has previously been done. Any time we specify a byte range
+        # we must disable the gzip compression on the server since we want
+        # to address ranges in the uncompressed file. If we send ranges that
+        # are interpreted by the server as offsets into the compressed file
+        # then that is at least confusing and also there is no guarantee that
+        # the bytes can be uncompressed.
 
         end_pos = self._current_position + (size - 1) if size >= 0 else ""
-        headers = {"Range": f"bytes={self._current_position}-{end_pos}"}
+        headers = {"Range": f"bytes={self._current_position}-{end_pos}", "Accept-Encoding": ""}
 
-        with time_this(self._log, msg="Read from remote resource %s", args=(self._url,)):
+        with time_this(
+            self._log, msg="Read from remote resource %s using headers %s", args=(self._url, headers)
+        ):
             resp = self._session.get(self._url, stream=False, timeout=self._timeout, headers=headers)
 
+        if resp.status_code == 416:
+            # Must have run off the end of the file. A standard file handle
+            # will treat this as EOF so be consistent with that.
+            self._current_position = int(end_pos)
```

Comment:
I dont think int(end_pos) here is correct, that is saying it is at the position past the end of the file, as if it could read that far. Is it true that every 416 will have what bytes it can read returned? I thought some would send back a 416 with no bytes read sometimes? If that is true, then you do need to ask again for the remaining bytes.

Reply:
I was trying to comply with what happens when you try to do this with a normal file handle. When  you call seek() off the end of the file and then call read() you end up with empty bytes as if EOF and tell() reports your seek position. Now that I write this I realize that not changing the current position at all would be consistent so I will remove this line.

Reply:
I am also worried about the case where:
1) the current position is at byte 10 of a 12 byte file
2) someone request a read of 4 bytes
3) a 416 is returned, but those last 2 bytes may not have been read.

Reply:
Ok, it seemed a little ambiguous when I read https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/416

Reply:
That does seem to be ambiguous but github and wsgidav are definitely letting you ask for 4 bytes but only returning 2.